storage/badger/chunkDataPacks.go

```diff
@@ -67,25 +71,63 @@ func (ch *ChunkDataPacks) Remove(chunkID flow.Identifier) error {
 }
 
 func (ch *ChunkDataPacks) BatchStore(c *flow.ChunkDataPack, batch storage.BatchStorage) error {
+	sc := toStoredChunkDataPack(c)
 	writeBatch := batch.GetWriter()
 	batch.OnSucceed(func() {
-		ch.byChunkIDCache.Insert(c.ChunkID, c)
+		ch.byChunkIDCache.Insert(sc.ChunkID, sc)
 	})
-	return operation.BatchInsertChunkDataPack(c)(writeBatch)
+	return operation.BatchInsertChunkDataPack(sc)(writeBatch)
 }
 
 func (ch *ChunkDataPacks) ByChunkID(chunkID flow.Identifier) (*flow.ChunkDataPack, error) {
 	tx := ch.db.NewTransaction(false)
 	defer tx.Discard()
-	return ch.retrieveCHDP(chunkID)(tx)
+
+	schdp, err := ch.retrieveCHDP(chunkID)(tx)
+	if err != nil {
+		return nil, fmt.Errorf("could not retrive stored chunk data pack: %w", err)
+	}
+
+	chdp := &flow.ChunkDataPack{
+		ChunkID:    schdp.ChunkID,
+		StartState: schdp.StartState,
+		Proof:      schdp.Proof,
+	}
+
+	if schdp.CollectionID != nil {
+		collection, err := ch.collections.ByID(*schdp.CollectionID)
+		if err != nil {
+			return nil, fmt.Errorf("could not retrive collection for stored chunk data pack: %w", err)
```

Comment:
better to include the collection id

Reply:
Fixed in https://github.com/onflow/flow-go/pull/968/commits/0a22d848633ab2ec942b347202d3e7b0fe40c7d9.